src/external_acl.cc

```diff
@@ -999,9 +1001,11 @@ externalAclHandleReply(void *data, const Helper::Reply &reply)
         entry = external_acl_cache_add(state->def, state->key, entryData);
 
     do {
-        void *cbdata;
-        if (state->callback && cbdataReferenceValidDone(state->callback_data, &cbdata))
-            state->callback(cbdata, entry);
+        CallBack(state->codeContext, [&] {
+            void *cbdata = nullptr;
+            if (state->callback && cbdataReferenceValidDone(state->callback_data, &cbdata))
+                state->callback(cbdata, entry);
+        });
 
```

Comment:
Squid helpers are used for more than just external ACL lookups, so these external_acl changes appear to be rather independent from the helper.cc changes that this PR focuses on. Perhaps we should split this PR into two?

A possibly related question is why external ACLs are so special in the context of this PR? Should not _all_ asynchronous ACLs have some kind of CodeContext saving/restoring code? And if yes, perhaps that code should live somewhere inside ACLChecklist::goAsync() and related asynchronous ACL code (instead of being duplicated in every async/slow ACL)??